packages/compass-sidebar/src/components/legacy/sidebar-databases-navigation.tsx

```diff
@@ -54,7 +56,11 @@ function SidebarDatabasesNavigation({
           openCollectionWorkspace(ns, { newTab: true });
           return;
         case 'modify-view': {
-          const coll = findCollection(ns, databases);
+          const coll = findCollection(
+            ns,
+            (connections.find((cn) => cn.connectionInfo.id === connectionId)
+              ?.databases as Database[]) || []
+          );
```

Comment:
since we're already working with a single connection, we could simplify -> 
```suggestion
          const coll = findCollection(
            ns, connection.databases as Database[],
          );
```